src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -50,23 +55,35 @@ public static void setHeader(final SAMFileHeader header) {
      * @param outputFile path to the output bam.
      * @param rddReads reads to write.
      * @param header the header to put at the top of the files
-     * @param singleFile should the output be a single file or sharded, i.e., outputfile/part-r-00000, outputfile/part-r-00001...
+     * @param format should the output be a single file, sharded, ADAM, etc.
      */
     public static void writeReads(
             final JavaSparkContext ctx, final String outputFile, final JavaRDD<GATKRead> rddReads,
-            final SAMFileHeader header, boolean singleFile) throws IOException {
-        if (singleFile) {
+            final SAMFileHeader header, ReadsWriteFormat format) throws IOException {
+        if (format.equals(ReadsWriteFormat.SINGLE)) {
             writeReadsSingle(ctx, outputFile, rddReads, header);
-        } else {
+        } else if (format.equals(ReadsWriteFormat.SHARDED)) {
             writeReadsSharded(ctx, outputFile, rddReads, header);
-
+        } else if (format.equals(ReadsWriteFormat.ADAM)) {
+            writeReadsADAM(ctx, outputFile, rddReads, header);
         }
+    }
 
+    private static void writeReadsADAM(
+            final JavaSparkContext ctx, final String outputFile, final JavaRDD<GATKRead> rddGATKReads,
+            final SAMFileHeader header) throws IOException {
+        JavaPairRDD<Void, AlignmentRecord> rddAlignmentRecords =
+                rddGATKReads.map(new GATKReadToBDGAlignmentRecordConverter(header)).mapToPair(
+                        alignmentRecord -> new Tuple2<Void, AlignmentRecord>(null, alignmentRecord));
+        Job job = Job.getInstance(ctx.hadoopConfiguration());
```

Comment:
Please a comment explaining what's happening here. Why do you need the job? What's AvroParquetOutputFormat.setSchema doing? Etc. saveAsNewAPIHadoopFile can be a little hard to reason about.